### PR TITLE
Prep docs for v9

### PR DESCRIPTION
**Pull request by jakelandis**
This commit prepares the docs for v9. 
* removes all v8 release notes, highlights, and breaking changes
* update v8 files to v9 as place holders for v9 docs
* updates references to current (9) and current - 1 (8) where appropriate
* ran ./gradlew generateReleaseNotes

Note - Some of this content is generated by `./gradlew generateReleaseNotes` and is may not be accurate to v9 only. We will need to manually adjust later. For now just trying to get the big change in. 

Also, the REST API doc is showing up in the transient setting migration guide ...I added a fixme to fix later. Again, just trying to get the big change in.

### CODE REVIEW

File: docs/reference/release-notes/highlights.asciidoc
Comment (jakelandis) [inline docs/reference/release-notes/highlights.asciidoc:39]: AFAIK these are all 8.x release highlights.

File: docs/reference/rest-api/rest-api-compatibility.asciidoc
Comment (jakelandis) [inline docs/reference/rest-api/rest-api-compatibility.asciidoc:41]: @jmikell821 - can you help me figure out how to make this reference work ?

**Comment by mark-vieira** [comment]
> Note - I think we will end up fighting with some of the automation for changes targeting both 8.x and 9.0.

I _think_ we've updated thing so we don't forward port 8.x release notes to `main` (9.0). @brianseeders can you confirm/deny this?

File: docs/reference/rest-api/rest-api-compatibility.asciidoc
Comment (leemthompo) [inline docs/reference/rest-api/rest-api-compatibility.asciidoc:41]:
```suggestion
//compatibility mode. See <<rest-api-changes,REST API changes>> in the migration guide.
```

driving by here :)

Comment (jakelandis) [reply docs/reference/rest-api/rest-api-compatibility.asciidoc:41]: Thanks Liam, I tried that but got:
```
INFO:build_docs:asciidoctor: WARNING: invalid reference: rest-api-changes
```
The link is trying :
docs/reference/rest-api/rest-api-compatibility.asciidoc <- from here
docs/reference/migration/migrate_9_0/rest-api-changes.asciidoc <- to here

This also, does not work: `See <<breaking_90_rest_api_changes,REST API changes>>` 
```
INFO:build_docs:asciidoctor: WARNING: invalid reference: breaking_90_rest_api_changes
```

I only updated the file name and anchor name to be `9` instead of `8`. This used to work : `<<breaking_80_rest_api_changes, REST API changes>>` 

So I assume that maybe I removed some implicit transitive include or something like that (I am pretty clueless with asciidoc)

**Comment by jakelandis** [comment]
> Note - I think we will end up fighting with some of the automation for changes targeting both 8.x and 9.0.

Since the process is to start with main and then backport, the automation will create the change logs in main. I'm not sure exactly how those are used to create the release notes, but it will likely require some manual effort to determine if those apply to 9.0.0 only or not. For example, since yesterday the following change logs have been created. There is at least 1 in there that is 9.0.0 only (as determined by the "type: breaking " label). 

image


I will take another look at the recent change logs removed here and preserve them (before i merge this PR if they have "type: breaking", but it will be very difficult without automation support to know based on the information in the change log if it is a 9.0.0 only or not change.

**Comment by jakelandis** [comment]
pull back to draft since I just realized this competes with https://github.com/elastic/elasticsearch/pull/114154 that does the same thing (but slightly better).

**Comment by jakelandis** [comment]
Closing this as dupcliate of https://github.com/elastic/elasticsearch/pull/114154

**Comment by jakelandis** [comment]
re-opening since  https://github.com/elastic/elasticsearch/pull/114154 was closed

**Comment by jakelandis** [comment]
@leemthompo - could you please review. the docs don't need to be perfect for now, but we need a place to start collecting the information.

**Review by leemthompo — COMMENTED** [review]
Couple questions, happy to approve if need unblocking to move forward, just re-ping :)

File: docs/reference/migration/migrate_9_0.asciidoc
Comment (jakelandis) [reply docs/reference/migration/migrate_9_0.asciidoc:2]: We will end fighting abit with the automation. So even if i clean this up now, it will just come back the next time someone generates the release notes. I think we will need to fix this is in the automation, or just go manual, or something in-between. I would prefer to address in a follow up since the changes here are intended to mechanical. I will add a comment to this file (and others) something to the effect that this is g

File: docs/reference/release-notes/highlights.asciidoc
Comment (jakelandis) [reply docs/reference/release-notes/highlights.asciidoc:2]: yeah, this was created by automation.  If if fix it now, it will come back the next time some generates the release notes. I will add a comment to this effect to this file too and will address in a follow up.

Comment (leemthompo) [reply docs/reference/release-notes/highlights.asciidoc:2]: thanks!

**Review by leemthompo — APPROVED** [review]

File: docs/reference/migration/migrate_9_0.asciidoc
Comment (leemthompo) [inline docs/reference/migration/migrate_9_0.asciidoc:1]: ❤️

File: docs/reference/migration/migrate_9_0.asciidoc
Comment (jakelandis) [reply docs/reference/migration/migrate_9_0.asciidoc:5]: I updated this PR with a comment for these generated file that links to an internal issue for us to follow up with. I also commented out the content created by the automation. While no one is probably looking at the v9 release notes (it's not released yet) it is better to have no information than wrong information.

Comment (leemthompo) [reply docs/reference/migration/migrate_9_0.asciidoc:5]: 💯

**Pull request merged**
